src/object/schema_manager.c

```diff
@@ -2867,23 +2873,10 @@ sm_rename_class (MOP class_mop, const char *new_name)
 	      tran_abort_upto_system_savepoint (UNIQUE_PARTITION_SAVEPOINT_RENAME);
 	    }
 
```

Comment:
```suggestion
db_private_free_and_init (NULL, class_new_name);
```

Reply:
After sm_flush_objects(), class_new_name must not be freed.